api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -221,6 +205,33 @@ public Encounter saveEncounter(Encounter encounter) throws APIException {
 		addGivenObsAndTheirGroupMembersToEncounter(obsToAdd, encounter);
 		return encounter;
 	}
+	
+	// if authenticated user is not supposed to edit encounter of certain type
```

Comment:
* same here please remove comment
* I think these private methods do not need to declare `throws APIException` since its a runtime exception